### PR TITLE
feat(PageHead): add noBorder prop

### LINKED ISSUE #2955: PageHead border needs a way of turning it off
**Component**
[PageHead](https://picasso.toptal.net/?path=/story/components-pagehead--pagehead)

**Is your feature request related to a problem? Please describe.**
For certain pages/conditions we want to be able to have a page header without bottom border — so it does not clash visually with the next blocks border.

**Describe the solution you'd like**
A `noBorder = false` prop that can be set to `true` when border is not desired. Implementation-wise, it would control if `root::after` style is applied.

**Describe alternatives you've considered**
Can't see any.

**Additional context**
image

### PR DESCRIPTION
**Pull request by TomasSlama**
[FX-NNNN]

### Description

closes #2955 

Users needs to be more flexible and hide bottom border of PageHead when directly followed by bordered section. Details and design approval is in the issue (#2955)

### How to test

- Check storybook
- Check new screenshot in Happo 🙇 

### Screenshots

image

### Development checks

- [x] Add changeset according to [guidelines](https://github.com/toptal/picasso/blob/master/docs/contribution/changeset-guidelines.md) (if needed)
- [x] Read [CONTRIBUTING.md](https://github.com/toptal/picasso/blob/master/CONTRIBUTING.md) and [Component API principles](https://github.com/toptal/picasso/blob/master/docs/contribution/component-api.md)
- [x] Make sure that additions and changes on design follow [Toptal's BASE design](https://design.toptal.net/), and it's been already discussed with designers at #-base-core
- [x] Annotate all `props` in component with documentation
- [x] Create `examples` for component
- [x] Ensure that deployed demo has expected results and good examples
- [x] Ensure the changed/created components have not caused accessibility issues. [How to use accessibility plugin in storybook](https://github.com/toptal/picasso/blob/master/docs/contribution/accessibility.md).
- [x] Self reviewed
- [x] Covered with tests

> All **_development checks_** should be done and set checked to pass the
> **GitHub Bot: TODOLess** action

<details>
<summary>PR commands</summary>
<br />

List of available commands:

- `@toptal-bot run package:alpha-release` - Release alpha version
- `@toptal-anvil ping reviewers` - Ping FX team for review

</details>

<details>
<summary>PR Review Guidelines</summary>
<br />

#### When to approve? ✅

**You are OK** with merging this PR and

1. You have no extra requests.
2. You have optional requests.
   1. Add `nit:` to your comment. (ex. `nit: I'd rename this variable from makeCircle to getCircle`)

#### When to request changes? ❌

**You are not OK** with merging this PR because

1. Something is broken after the changes.
2. Acceptance criteria is not reached.
3. Code is dirty.

#### When to comment (neither ✅ nor ❌)

**You want your comments to be addressed** before merging this PR in cases like:

1. There are leftovers like unnecessary logs, comments, etc.
2. You have an opinionated comment regarding the code that requires a discussion.
3. You have questions.

#### How to handle the comments?

1. An owner of a comment is the only one who can resolve it.
2. An owner of a comment must resolve it when it's addressed.
3. A PR owner must reply with ✅ when a comment is addressed.

</details>

### CODE REVIEW

**Comment by TomasSlama** [comment]
@toptal-anvil ping reviewers

File: .changeset/hungry-clouds-stare.md [hunk 1]
```diff
@@ -0,0 +1,10 @@
+---
+'@toptal/picasso': minor
+---
+
+---
+
+### PageHead
+
+- add `noBorder` property to hide bottom border, when PageHead is directly followed by bordered section
+- check [storybook example](https://picasso.toptal.net/?path=/story/components-pagehead--pagehead#no-border) for more details
```
Comment (augustobmoura) [inline .changeset/hungry-clouds-stare.md:10]: NIT: I would put these two lines in the same bullet point, as they mention the same feature. I think it looks better on the changelogs

```suggestion
- add `noBorder` property to hide bottom border, when PageHead is directly followed by bordered section.
  Check [storybook example](https://picasso.toptal.net/?path=/story/components-pagehead--pagehead#no-border) for more details
```

**Review by augustobmoura — APPROVED** [review]
LGTM! :+1:

**Review by OndrejTuma — APPROVED** [review]
Code looks good 👌 
Tested in temploy

**Review by ozgurkececioglu — APPROVED** [review]
LGTM 👍

File: .changeset/hungry-clouds-stare.md [hunk 2]
```diff
@@ -0,0 +1,10 @@
+---
+'@toptal/picasso': minor
+---
+
+---
+
+### PageHead
+
+- add `noBorder` property to hide bottom border, when PageHead is directly followed by bordered section.
```
Comment (falsefalse) [inline .changeset/hungry-clouds-stare.md:9]: ❤️

**Review by falsefalse — APPROVED** [review]
Thank you so much 🙇  Can't wait for client portal to bump picasso version

**Pull request merged**